### DISCUSSION TITLE
Best way to include opti variables inside of parameters of differential equations

### DISCUSSION THREAD

**Discussion by akb427** [casadi/casadi#3403] [Q&A] [answered]
Hello,

I am trying to solve an optimization problem with an equation of the form \dot{x} = A(m)*x+Bu, where m is a time series of opti variables (with additional constraints placed on them), and a range of acceptable x values for all time steps. What would be the best way to formulate this optimization problem?

My initial attempt was to calculate A using a single time step of m. Then I would run a for loop with a forward euler approximation of x with a much smaller time step. This was to ensure that the dynamic equation remained stable. However this seems very slow and clunky, as the final form of \dot{x} may not be easily differentiated by ipopt.

My next throught was using the matrix exponential, but was having trouble getting it to work in matlab. It seems like the Taylor Sieres approximation of the matrix exponential would be essentially the same as the for loop implementation.

Is there a way to use the integrators function to do this? I am new to CasADi and was having trouble figuring them out.

Am I missing another option? I am looking for efficiency since this code needs to be solved thousands of times.

**Comment by jgillis** [comment, marked as answer]
One way to look at this, is noting that, as the integrator is concerned, both m and u are time-varying inputs, no? If you call the integrator with opti variables, parameters or numbers does not matter.
